ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx

```diff
@@ -118,8 +121,9 @@ export function HistoryDialogContainer(props: HistoryDialogContainerProps) {
         activeItem: version
       });
     },
-    [item?.systemType, item?.availableActionsMap.revert, count, setMenu, formatMessage, isConfig, item?.stateMap.locked]
+    [item, count, setMenu, formatMessage, isConfig]
```

Comment:
The item object can change frequently. Why not keep the individual properties instead of the instance as dependencies of the callback?

Reply:
I need the whole object a couple of lines before when I do `isPreviewable(item)`